src/arrayMethodJoin.js

```diff
@@ -4,8 +4,32 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let joinedArray = '';
+
+    for (const item of this) {
+      if (item === null || item === undefined) {
+        joinedArray += '' + separator;
```

Comment:
joinedArray is already a string.
```suggestion
        joinedArray += separator;
```